addons/mrp_subcontracting/models/mrp_bom.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class MrpBom(models.Model):
+    _inherit = 'mrp.bom'
+
+    type = fields.Selection(selection_add=[('subcontract', 'Subcontracting')])
+    subcontractor_ids = fields.One2many('res.partner', 'bom_id', domain=[('type', '=', 'subcontractor')], string='Subcontractors')
+
+    def _bom_subcontract_find (self, product_tmpl=None, product=None, picking_type=None, company_id=False, bom_type=False, subcontractor=False):
+        domain = self._bom_find_domain(product_tmpl=product_tmpl, product=product, picking_type=picking_type, company_id=company_id, bom_type=bom_type)
+        if domain is False:
```

Comment:
_bom_subcontract_find could remove the kwarg bom_type and direclty replace it with subcontract